src/wp-includes/class-wp-scripts.php

```diff
@@ -512,31 +476,125 @@ public function add_inline_script( $handle, $data, $position = 'after', $standal
 	 * @return string|false Script on success, false otherwise.
 	 */
 	public function print_inline_script( $handle, $position = 'after', $display = true ) {
-		$output = $this->get_data( $handle, $position );
-
+		// TODO: Add _deprecated_function() call. There are only 11 plugins that call this function, and they almost all pass $display=false. See https://wpdirectory.net/search/01H1QQF0S8RPH5AQR6YMTV0RHV.
+		if ( $display ) {
+			$output = $this->get_inline_script_tag( $handle, $position );
+			echo $output;
+		} else {
+			$output = $this->get_inline_script_data( $handle, $position );
+		}
 		if ( empty( $output ) ) {
 			return false;
 		}
+		return $output;
+	}
+
+	/**
+	 * Gets data for inline scripts registered for a specific handle.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
+	 * @param string $position Optional. Whether to add the inline script
+	 *                         before the handle or after. Default 'after'.
+	 * @return string Inline script, which may be empty string.
+	 */
+	public function get_inline_script_data( $handle, $position ) {
+		$data = $this->get_data( $handle, $position );
+		if ( empty( $data ) || ! is_array( $data ) ) {
+			return '';
+		}
 
-		$output = trim( implode( "\n", $output ), "\n" );
+		return trim( implode( "\n", $data ), "\n" );
+	}
 
-		if ( $display ) {
-			if ( 'after-non-standalone' === $position ) {
-				$script_output = "<script%1\$s id='%2\$s-js-after' data-wp-executes-after='%2\$s'>\n%4\$s\n</script>\n";
-			} else {
-				$script_output = "<script%1\$s id='%2\$s-js-%3\$s'>\n%4\$s\n</script>\n";
-			}
+	/**
+	 * Gets tags for inline scripts registered for a specific handle.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
+	 * @param string $position Optional. Whether to add the inline script
+	 *                         before the handle or after. Default 'after'.
+	 * @return string Inline script, which may be empty string.
+	 */
+	public function get_inline_script_tag( $handle, $position ) {
+		$js = $this->get_inline_script_data( $handle, $position );
+		if ( empty( $js ) ) {
+			return '';
+		}
+
+		// TODO: Handle case where a dep has a src which is false.
+		$id   = "{$handle}-js-{$position}";
+		$deps = $this->registered[ $handle ]->deps;
 
-			printf(
-				$script_output,
-				$this->type_attr,
-				esc_attr( $handle ),
-				esc_attr( $position ),
-				$output
+		if ( $this->should_delay_inline_script( $handle, $position ) ) {
+			$attributes = array(
+				'id'   => $id,
+				'type' => 'text/template', // TODO: Consider text/plain instead.
```

Comment:
@joemcgill @10upsimon Why was `text/template` initially chosen for the script type? Note that [`amp-script` uses `text/plain` for such scripts](https://amp.dev/documentation/components/amp-script#from-a-local-element). I think this would be better as there's nothing really "templatey" about such inline scripts.

Reply:
I don't think `text/plain` was ever considered as an alternative, but I agree that it's a better semantic choice as long as there are no negative side effects to making this change.

Reply:
The only requirement is that it not be `text/javascript` or another JavaScript-ish content type to prevent it from being evaluated by the browser. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type), its value may be:

> ![image](https://github.com/10up/wordpress-develop/assets/134745/afa7ac40-4cb3-4436-a439-72101862bf1f)

The [JavaScript MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#textjavascript) are:

> ![image](https://github.com/10up/wordpress-develop/assets/134745/a1a44c28-d0cd-4da1-973b-87f289d6aae9)

So as long as we don't use one of those, or `module` nor `importmap`, then it is considered a data block. It says we're supposed to use a valid MIME type, and `text/template` actually isn't one. However, [`text/plain`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#textplain) is a valid MIME type. Also, considering much thought was put into its choice for `amp-script`, it seems like the best.

Reply:
Implemented in c164071

Reply:
@westonruter thanks for this. I looked back over some initial comms, there was no specific reason for choosing `text/template`, it appears it may have been an early decision made against better judgement.